src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -695,6 +697,26 @@ public function appendToFile(string $filename, $content)
         }
     }
 
+    /**
+     * Watches a file or directory for any changes, and calls $callback when any changes are detected.
+     *
+     * @param mixed    $path     The path to watch for changes. Can be a path to a file or directory, iterator or array with paths
+     * @param callable $callback The callback to execute when a change is detected
+     * @param float    $timeout  The idle timeout in milliseconds after which the process will be aborted if there are no changes detected
+     *
+     * @throws \InvalidArgumentException|IOException
+     */
+    public function watch($path, callable $callback, float $timeout = null)
```

Comment:
It would be nice to define events we want to listen (by default creating, changing, and deleting).

Reply:
The events are passed to the callback function, so you can skip certain events already by adding an `if` statement in the callback

Reply:
Some instructions take time to be executed and limit changes we want to watch can help to reduce useless time. Also, the change must in the watcher code to do it.